packages/react-events/src/dom/Tap.js

```diff
@@ -0,0 +1,615 @@
+    pressure,
+    screenX,
+    screenY,
+    shiftKey,
+    tangentialPressure,
+    tiltX,
+    tiltY,
+    twist,
+    width,
+    clientX,
+    clientY,
+  } = nativeEvent;
+
+  return {
+    altKey,
+    buttons: state.buttons,
+    ctrlKey,
+    height,
+    metaKey,
+    pageX,
+    pageY,
+    pointerType,
+    pressure,
+    screenX,
+    screenY,
+    shiftKey,
+    tangentialPressure,
+    target: event.responderTarget,
+    tiltX,
+    tiltY,
+    timeStamp,
+    twist,
+    width,
+    x: clientX,
+    y: clientY,
+  };
+}
+
+function createFallbackGestureState(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+  event: ReactDOMResponderEvent,
+): TapGestureState {
+  const timeStamp = context.getTimeStamp();
+  const nativeEvent = (event.nativeEvent: any);
+  const isUpEvent = event.type === 'mouseup' || event.type === 'touchend';
+  const isCancelType =
+    event.type === 'dragstart' || event.type === 'touchcancel';
+  const isTouchEvent = nativeEvent.changedTouches != null;
+
+  const {altKey, ctrlKey, metaKey, shiftKey} = nativeEvent;
+
+  let pointerEvent = nativeEvent;
+  if (isTouchEvent) {
+    // TODO: use getTouchById
+    pointerEvent = nativeEvent.changedTouches[0];
+  }
+
+  const {
+    pageX,
+    pageY,
+    radiusX,
+    radiusY,
+    rotationAngle,
+    screenX,
+    screenY,
+    clientX,
+    clientY,
+  } = pointerEvent;
+
+  return {
+    altKey,
+    buttons: state.buttons != null ? state.buttons : 1,
+    ctrlKey,
+    height: !isCancelType && radiusY != null ? radiusY * 2 : 1,
+    metaKey,
+    pageX: isCancelType ? 0 : pageX,
+    pageY: isCancelType ? 0 : pageY,
+    pointerType: event.pointerType,
+    pressure: isUpEvent || isCancelType ? 0 : isTouchEvent ? 1 : 0.5,
+    screenX: isCancelType ? 0 : screenX,
+    screenY: isCancelType ? 0 : screenY,
+    shiftKey,
+    tangentialPressure: 0,
+    target: event.responderTarget,
+    tiltX: 0,
+    tiltY: 0,
+    timeStamp,
+    twist: rotationAngle != null ? rotationAngle : 0,
+    width: !isCancelType && radiusX != null ? radiusX * 2 : 1,
+    x: isCancelType ? 0 : clientX,
+    y: isCancelType ? 0 : clientY,
+  };
+}
+
+const createGestureState = hasPointerEvents
+  ? createPointerEventGestureState
+  : createFallbackGestureState;
+
+/**
+ * Managing root events
+ */
+
+function addRootEventTypes(
+  rootEvents: Array<string>,
+  context: ReactDOMResponderContext,
+  state: TapState,
+): void {
+  if (!state.rootEvents) {
+    state.rootEvents = rootEvents;
+    context.addRootEventTypes(state.rootEvents);
+  }
+}
+
+function removeRootEventTypes(
+  context: ReactDOMResponderContext,
+  state: TapState,
+): void {
+  if (state.rootEvents != null) {
+    context.removeRootEventTypes(state.rootEvents);
+    state.rootEvents = null;
+  }
+}
+
+/**
+ * Managing pointers
+ */
+
+function isActivePointer(event, state) {
+  return (
+    state.pointerType === event.pointerType &&
+    state.pointerId === event.pointerId
```

Comment:
This check is broken because `event.pointerId` is only populated for PointerEvent